tips/TIP-0020/tip-0020.md

```diff
@@ -0,0 +1,489 @@
+
+#### Inputs
+
+The `Inputs` field holds the inputs to consume in order to fund the outputs of the <i>Transaction Payload</i>. Currently, there is only one type of input, the <i>UTXO Input</i>. In the future, more types of inputs may be specified as part of protocol upgrades.
+
+Each input must be accompanied by a corresponding <i>Unlock</i> at the same index in the <i>Unlocks</i> part of the <i>Transaction Payload</i>.
+
+##### UTXO Input
+
+A <i>UTXO Input</i> is an input which references an unspent output of a previous transaction. This UTXO is uniquely identified by its _Output ID_, defined by the _Transaction ID_ of the creating transaction together with corresponding output index. Each <i>UTXO Input</i> must be accompanied by an <i>Unlock</i> that is allowed to unlock the referenced output.
+
+#### Inputs Commitment
+
+The _Inputs Commitment_ field of the _Transaction Essence_ is a cryptographic commitment to the content of the consumed outputs (inputs). The transaction itself only references inputs by _Output ID_, therefore the client has to be aware of the content to produce a semantically valid transaction. The commitment is a security mechanism that protects clients against [eclipse attacks that would result in loss of funds.](https://github.com/iotaledger/tips/discussions/51)
+
+#### Outputs
+
+The `Outputs` field holds the outputs that are created by the <i>Transaction Payload</i>. There are different output types, but they must all have an `Amount` field denoting the number of IOTA coins to deposit.
+
+The following table lists all the output types that are currently supported as well as links to the corresponding specification. The _SigLockedSingleOutput_ as well as the _SigLockedDustAllowanceOutput_ introduced in Chrysalis Phase 2 [TIP-7](https://github.com/iotaledger/tips/blob/main/tips/TIP-0007/tip-0007.md) have been removed and are no longer supported.
+
+| Output Name | Type Value | TIP                                                                                                          |
+| ----------- | ---------- | ------------------------------------------------------------------------------------------------------------ |
+| Basic    | 3          | [draft TIP-18](https://github.com/lzpap/protocol-rfcs/blob/master/tips/TIP-0018/tip-0018.md#basic-output) |
+| Alias       | 4          | [draft TIP-18](https://github.com/lzpap/protocol-rfcs/blob/master/tips/TIP-0018/tip-0018.md#alias-output)    |
+| Foundry     | 5          | [draft TIP-18](https://github.com/lzpap/protocol-rfcs/blob/master/tips/TIP-0018/tip-0018.md#foundry-output)  |
+| NFT         | 6          | [draft TIP-18](https://github.com/lzpap/protocol-rfcs/blob/master/tips/TIP-0018/tip-0018.md#nft-output)      |
```

Comment:
What do we do here? Leave TIP-18 as a draft pointing to the PR and then change it to the proper link, once that gets merged?

Reply:
Yes, my plan is to merge all the TIPs and then create PR that updates the refs present in the repo.